src/batch-builder.js

```diff
@@ -394,37 +381,44 @@ module.exports = class BatchBuilder {
             this._accumulateFees(tx.tokenID, fee2Charge);
         }
 
-        if (tx.fromIdx === finalToIdx)
+        if (tx.fromIdx === finalToIdx || isExit)
             oldState2 = Object.assign({}, newState1);
 
         let newState2;
         if (op2 != "NOP"){
             newState2 = Object.assign({}, oldState2);
-            newState2.balance = Scalar.add(oldState2.balance, effectiveAmount);
-        }
 
-        if (op1=="INSERT") {
+            if (isExit){
+                newState2.exitBalance = Scalar.add(oldState2.exitBalance, effectiveAmount);
+            } else {
+                newState2.balance = Scalar.add(oldState2.balance, effectiveAmount);
+                newState2.accumulatedHash = stateUtils.computeAccumulatedHash(oldState2.accumulatedHash, tx, this.nLevels);
```

Comment:
accumulate hash should take in account if the amount is or not nullified in case of L1!

Reply:
good catch !